src/_P021_Level.ino

```diff
@@ -60,70 +159,171 @@ boolean Plugin_021(uint8_t function, struct EventStruct *event, String& string)
       break;
     }
 
+    case PLUGIN_GET_DEVICEVALUECOUNT:
+    {
+      event->Par1 = P021_getConfigBit(P021_FLAGS, P021_STATE_OUTP) ? 2 : 1;
+      success     = true;
+      break;
+    }
+
     case PLUGIN_GET_DEVICEVALUENAMES:
     {
       strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[0], PSTR(PLUGIN_VALUENAME1_021));
+      strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[1], PSTR(PLUGIN_VALUENAME2_021));
       break;
     }
 
     case PLUGIN_GET_DEVICEGPIONAMES:
     {
-      event->String1 = formatGpioName_output(F("Level low"));
+      event->String1 = formatGpioName_output(F("Output"));
       break;
     }
 
     case PLUGIN_SET_DEFAULTS:
     {
-      P021_DONT_ALWAYS_SAVE = 1; // Do not save
+      P021_DONT_ALWAYS_SAVE = 1;                   // Do not save
+      P021_GPIO_RELAY       = -1;                  // GPIO for relay output not assigned
+      P021_CHECK_TASK       = -1;                  // No input source assigned
+      P021_CHECK_VALUE      = -1;                  // No input source assigned
+      P021_SETPOINT         = 25.0f;               // Switch output active above 25 [deg C]
+      P021_HYSTERESIS       = 5.0f;                // Switch off hysteresis 5 [deg C] below switch on value
+      P021_MIN_TIME         = 30;                  // Once switched on output should be active at least 30 [min]
+      P021_INTERVAL_TIME    = 24;                  // Output shall run after 24 [hour] stand still
+      P021_FORCE_TIME       = 5;                   // Forced circulation for 5 [min]
+      P021_OPMODE           = P021_OPMODE_OFF;     // Don't touch output unless selected by operator
+      P021_FLAGS            = 0;                   // Reset all flags
+      UserVar.setFloat(event->TaskIndex, P021_VALUE_STATE, (float)P021_STATE_IDLE);
+      P021_remote[event->TaskIndex]    = 0;        // Remote control state is "off"
+      P021_timestamp[event->TaskIndex] = millis(); // Start counting time from now on
       break;
     }
 
     case PLUGIN_WEBFORM_LOAD:
     {
-      addRowLabel(F("Check Task"));
-      addTaskSelect(F("ptask"), P021_CHECK_TASK);
+       #ifdef PLUGIN_021_DEBUG
+      {
+        P021_control_state control_state = (P021_control_state)UserVar.getFloat(event->TaskIndex, P021_VALUE_STATE);
+
+        // Add some debug information
+        String msg = F("State= ");
+        msg += P021_printControlState(control_state);
+        msg += (F(", Pump= "));
+        msg += (int)UserVar.getFloat(event->TaskIndex, P021_VALUE_OUTPUT) == 1 ? F("on") : F("off");
+        msg += F(", Remote= ");
+        msg += P021_remote[event->TaskIndex];
+        msg += F(", Timer= ");
+        msg += (int)(millis2seconds(timePassedSince(P021_timestamp[event->TaskIndex])));
+        addFormNote(msg);
```

Comment:
Please use `strformat()` function to build complex (log) strings.

Reply:
Also you should not use very use case specific terms in this plugin.
The level plugin should not care/know whether it is driving a pump, some motor, a (water) valve, or anything else.
So better change "Pump" into "Output".

Reply:
Thanks for spotting some left-overs.